includes/fraud-prevention/class-order-fraud-and-risk-meta-box.php

```diff
@@ -0,0 +1,129 @@
+use WC_Payments_Utils;
+use WCPay\Constants\Fraud_Outcome_Status;
+
+/**
+ * Class Order_Fraud_And_Risk_Meta_Box
+ */
+class Order_Fraud_And_Risk_Meta_Box {
+	/**
+	 * The Order Service.
+	 *
+	 * @var WC_Payments_Order_Service
+	 */
+	private $order_service;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param WC_Payments_Order_Service $order_service The order service.
+	 */
+	public function __construct( WC_Payments_Order_Service $order_service ) {
+		add_action( 'add_meta_boxes', [ $this, 'maybe_add_meta_box' ] );
+
+		$this->order_service = $order_service;
+	}
+
+	/**
+	 * Maybe add the meta box.
+	 */
+	public function maybe_add_meta_box() {
+		// If fraud settings are off, or if we cannot get the screen ID, exit.
+		if ( ! WC_Payments_Features::is_fraud_protection_settings_enabled() || ! function_exists( '\wc_get_page_screen_id' ) ) {
+			return;
+		}
+
+		// Get the order edit screen to be able to add the meta box to.
+		$wc_screen_id = \wc_get_page_screen_id( 'shop-order' );
+
+		add_meta_box( 'wcpay_order_fraud_and_risk_meta_box', __( 'Fraud &amp; Risk', 'woocommerce-payments' ), [ $this, 'display_order_fraud_and_risk_meta_box_message' ], $wc_screen_id, 'side', 'default' );
+	}
+
+	/**
+	 * Displays the contents of the Fraud & Risk meta box.
+	 *
+	 * @param \WC_Order $order The order we are working with.
+	 *
+	 * @return void
+	 */
+	public function display_order_fraud_and_risk_meta_box_message( $order ) {
+		$order = wc_get_order( $order );
+
+		if ( ! $order ) {
+			return;
+		}
+
+		$intent_id      = $this->order_service->get_intent_id_for_order( $order );
+		$charge_id      = $this->order_service->get_charge_id_for_order( $order );
+		$outcome_status = $this->order_service->get_fraud_outcome_status_for_order( $order );
+
+		if ( 'woocommerce_payments' !== $order->get_payment_method() ) {
+			$outcome_status = 'not_wcpay';
+		}
+
+		switch ( $outcome_status ) {
+			case Fraud_Outcome_Status::ALLOW:
+				$icon_url    = plugins_url( 'assets/images/icons/check-green.svg', WCPAY_PLUGIN_FILE );
+				$icon_alt    = __( 'Green check mark', 'woocommerce-payments' );
+				$status      = __( 'No action taken', 'woocommerce-payments' );
+				$description = __( 'The payment for this order passed your risk filtering.', 'woocommerce-payments' );
+				echo '<p class="wcpay-fraud-risk-meta-allow"><img src="' . esc_url( $icon_url ) . '" alt="' . esc_html( $icon_alt ) . '"> ' . esc_html( $status ) . '</p><p>' . esc_html( $description ) . '</p>';
+				break;
+
+			case Fraud_Outcome_Status::REVIEW:
+				$icon_url        = plugins_url( 'assets/images/icons/shield-stroke-orange.svg', WCPAY_PLUGIN_FILE );
+				$icon_alt        = __( 'Orange shield outline', 'woocommerce-payments' );
+				$status          = __( 'Held for review', 'woocommerce-payments' );
+				$description     = __( 'The payment for this order was held for review by your risk filtering. You can review the details and determine whether to approve or block the payment.', 'woocommerce-payments' );
```

Comment:
Would the description be better phrased as:

> The payment for this order was held for review by your risk filtering. You can review the details and determine whether to capture the charge or cancel the authorization.

This will fall in line with the actions in the _Order actions_ menu.

Reply:
Are the merchants more familiar with 'Capture' or 'Approve' in terms of UI? The option on the orders screen actions select says 'Approve', and the transaction details page also contains the button with 'Approve' text in it.

Reply:
I think they will be more familiar with "Capture charge" and "Cancel authorization". The default text in the Order Edit screen where the meta box is going to appear is what's used here:

```
		$new_actions = [
			'capture_charge'       => __( 'Capture charge', 'woocommerce-payments' ),
			'cancel_authorization' => __( 'Cancel authorization', 'woocommerce-payments' ),
		];
```

https://github.com/Automattic/woocommerce-payments/blob/57a14b1c4214cb3fab2181c55fc9feda8ac50dcd/includes/class-wc-payment-gateway-wcpay.php#L2415-L2418

If we're moving away from this wording and are wanting to use something else, then maybe the options should be updated.

Reply:
Ok then, what you suggested looks good to me.